javaServices/coreJavaServices/src/joliex/util/TimeService.java

```diff
@@ -415,4 +423,44 @@ public Long getTimestampFromString( Value request )
 			throw new FaultException( "InvalidTimestamp", pe );
 		}
 	}
+
+	@RequestResponse
+	public Long scheduleTimeout( Value request )
+	{
+		final long timeoutId = atomicLong.getAndIncrement();
+		TimeUnit unit;
+
+		if (request.hasChildren( "timeunit" )) {
+			try {
+				unit = TimeUnit.valueOf( request.getFirstChild( "timeunit" ).strValue().toUpperCase() );
+			} catch ( Exception e ) {
+				System.err.println( e );
+				System.err.println( "Defaulting to MILLISECONDS" );
+				unit = TimeUnit.MILLISECONDS;
+			}
+		} else {
+			unit = TimeUnit.MILLISECONDS;
+		}
+
+		ScheduledFuture scheduledFuture = executor.schedule( () -> {
+				sendMessage( CommMessage.createRequest(request.getFirstChild( "operation" ).strValue(), "/", request.getFirstChild( "message" )));
+		}, request.intValue(), unit );
+		scheduledFutureHashMap.put(timeoutId, scheduledFuture);
+		return timeoutId;
+	}
+
+	public void cancelTimeout( Value request )
+	{
+		long timeoutId = request.longValue();
+		synchronized (lock) {
+			ScheduledFuture scheduledFuture = scheduledFutureHashMap.get( timeoutId );
+			if ( scheduledFuture != null ) {
+				scheduledFutureHashMap.get(timeoutId).cancel(false);
+				scheduledFutureHashMap.remove(timeoutId);
+			}
+			else {
```

Comment:
This makes the operation fail silently without the client knowing.

I think that it makes sense not to throw an exception in case the timer has already been cancelled, since execution of timers is concurrent and the client cannot really know if the timer expired already or not.

I would return a boolean value, though: `true` if the timer has been cancelled, `false` otherwise.

@jolie/developers comments?